joint_state_publisher:
  type: joint_state_controller/JointStateController
  publish_rate: 50.0

# controllers ---------------------------------------
reaction_wheel_position_controller:
  type: effort_controllers/JointPositionController
  joint: reaction_wheel_joint
  pid: {p: 1.0, i: 0.0, i_clamp: 0.0, d: 0.1}

reaction_wheel_velocity_controller:
  type: effort_controllers/JointVelocityController
  joint: reaction_wheel_joint
  pid: {p: -0.05, i: 0.0, i_clamp: 0.05, d: 0.0}

reaction_wheel_effort_controller:
  type: effort_controllers/JointEffortController
  joint: reaction_wheel_joint
  pid: {p: 1.0, i: 0.0, i_clamp: 0.0, d: 0.0}

left_shoulder_position_controller:
  type: effort_controllers/JointPositionController
  joint: left_shoulder
  pid: {p: 0.1, i: 0.0, i_clamp: 0.0, d: 0.02}

left_elbow_position_controller:
  type: effort_controllers/JointPositionController
  joint: left_elbow
  pid: {p: 0.1, i: 0.0, i_clamp: 0.0, d: 0.02}

left_wrist_position_controller:
  type: position_controllers/JointPositionController
  joint: left_wrist

# Trajectory controller ------------------------------
left_manipulator_trajectory_controller:
  type: "effort_controllers/JointTrajectoryController"
  joints:
      - left_shoulder
      - left_elbow
  gains: 
    left_shoulder: {p: 0.1, i: 0.0, i_clamp: 0.0, d: 0.01}
    left_elbow: {p: 0.1, i: 0.0, i_clamp: 0.0, d: 0.01}

# Gripper action Controller --------------------------
left_gripper_position_controller:
  type: "position_controllers/GripperActionController"
  joint: left_finger_joint
